module/dice/d20-roll.mjs

```diff
@@ -1,5 +1,34 @@
 const { Die, NumericTerm, OperatorTerm } = foundry.dice.terms;
 
+/**
+ * Configuration data for the process of rolling a d20 rolls.
+ *
+ * @typedef {BasicRollProcessConfiguration} D20RollProcessConfiguration
+ * @property {boolean} elvenAccuracy         Use three dice when rolling with advantage.
+ * @property {boolean} halflingLucky         Add a re-roll once modifier to the d20 die.
+ * @property {D20RollConfiguration[]} rolls  Configuration data for individual rolls.
+ */
+
+/**
+ * D20 roll configuration data.
+ *
+ * @typedef {BasicRollConfiguration} D20RollConfiguration
+ * @property {string[]} parts          Parts used to construct the roll formula, not including the d20 die.
+ * @property {D20RollOptions} options  Options passed through to the roll.
+ */
+
+/**
+ * Options that describe a d20 roll.
+ *
+ * @typedef {BasicRollOptions} D20RollOptions
+ * @property {boolean} advantage       Is the roll granted advantage?
+ * @property {boolean} disadvantage    Is the roll granted disadvantage?
+ * @property {number} criticalSuccess  The value of the d20 die to be considered a critical success.
+ * @property {number} criticalFailure  The value of the d20 die to be considered a critical failure.
+ * @property {number} maximum          Maximum number the d20 die can roll.
+ * @property {number} minimum          Minimum number the d20 die can roll.
```

Comment:
My kingdom for an additional `modifiers` property here.

Reply:
Adding this would be enshrining it as an officially supported part of the API. I don't think we can guarantee that any arbitrary `modifiers` would work with all combinations of other options that also add modifiers like `advantage`, `minimum`, `halflingLucky` etc., and with any other options we want to add in the future.

Reply:
We have a utility function in `dnd5e.utils` that could be used to clean the Set.